components/eam/src/physics/crm/physpkg.F90

```diff
@@ -1,3033 +1,1829 @@
-    ! Check energy integrals
-    call check_energy_chng(state, tend, "iondrag", nstep, ztodt, zero, zero, zero, zero)
-    call t_stopf  ( 'iondrag' )
+  ! store dse after tphysac in buffer
+  do k = 1,pver
+    dtcore(:ncol,k) = state%t(:ncol,k)
+  end do
 
-end if ! l_gw_drag
+  call set_dry_to_wet(state)    ! Physics had dry, dynamics wants moist
 
+  ! Scale dry mass and energy (does nothing if dycore is EUL or SLD)
+  call cnst_get_ind('CLDLIQ', ixcldliq)
+  call cnst_get_ind('CLDICE', ixcldice)
+  tmp_q     (:ncol,:pver) = state%q(:ncol,:pver,1)
+  tmp_cldliq(:ncol,:pver) = state%q(:ncol,:pver,ixcldliq)
+  tmp_cldice(:ncol,:pver) = state%q(:ncol,:pver,ixcldice)
+  call physics_dme_adjust(state, tend, qini, ztodt)
 
-if (l_ac_energy_chk) then
-    !-------------- Energy budget checks vvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvv
+  call diag_phys_tend_writeout (state, pbuf,  tend, ztodt, tmp_q, tmp_cldliq, tmp_cldice, tmp_t, qini, cldliqini, cldiceini)
 
-    call pbuf_set_field(pbuf, teout_idx, state%te_cur, (/1,itim_old/),(/pcols,1/))       
+  !-----------------------------------------------------------------------------
+  ! Set the chlorine and bromine mass mixing ratios
+  !-----------------------------------------------------------------------------
+  call clybry_fam_set( ncol, lchnk, map2chm, state%q, pbuf )
 
-    tmp_t(:ncol,:pver) = state%t(:ncol,:pver)
+  !-----------------------------------------------------------------------------
+  !Integrate column static energy and water vapor
+  !-----------------------------------------------------------------------------
+  call pbuf_get_field(pbuf, pbuf_get_index('static_ener_ac'), static_ener_ac_2d )
+  call pbuf_get_field(pbuf, pbuf_get_index('water_vap_ac'), water_vap_ac_2d )
 
-    ! store dse after tphysac in buffer
+  static_ener_ac_2d(:) = 0
+  water_vap_ac_2d(:)   = 0
+  do i = 1,ncol
     do k = 1,pver
-       dtcore(:ncol,k) = state%t(:ncol,k)
-    end do
-
-    !
-    ! FV: convert dry-type mixing ratios to moist here because physics_dme_adjust
-    !     assumes moist. This is done in p_d_coupling for other dynamics. Bundy, Feb 2004.
-
-    if ( dycore_is('LR') .or. dycore_is('SE')) call set_dry_to_wet(state)    ! Physics had dry, dynamics wants moist
-
-    ! Scale dry mass and energy (does nothing if dycore is EUL or SLD)
-    call cnst_get_ind('CLDLIQ', ixcldliq)
-    call cnst_get_ind('CLDICE', ixcldice)
-    tmp_q     (:ncol,:pver) = state%q(:ncol,:pver,1)
-    tmp_cldliq(:ncol,:pver) = state%q(:ncol,:pver,ixcldliq)
-    tmp_cldice(:ncol,:pver) = state%q(:ncol,:pver,ixcldice)
-    call physics_dme_adjust(state, tend, qini, ztodt)
-!!!   REMOVE THIS CALL, SINCE ONLY Q IS BEING ADJUSTED. WON'T BALANCE ENERGY. TE IS SAVED BEFORE THIS
-!!!   call check_energy_chng(state, tend, "drymass", nstep, ztodt, zero, zero, zero, zero)
-
-    !-------------- Energy budget checks ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
-end if ! l_ac_energy_chk
-
-    if (aqua_planet) then
```

Comment:
how come MMF-aqua planet didn't make it in the clean-up version of tphysac?

Reply:
Im pretty sure I determined that this check happened elsewhere, so it was redundant. I forget exactly where it happens though.